Correctly determine if a local group exists.

rpluem-vf opened the pull request:
##### SUMMARY
Fixes #58619 with the approach proposed by samdoran.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
group

##### ADDITIONAL INFORMATION

rpluem-vf committed:
Correctly determine if a local group exists.

Fixes #58619 with the approach proposed by samdoran.

rpluem-vf commented:
ready_for_review

rpluem-vf commented:
ready_for_review

rpluem-vf commented:
Ready_for_review

rpluem-vf commented:
Ready_for_review

rpluem-vf commented:
Anything I can do to move this forward?

rpluem-vf commented:
Ready_for_review

rpluem-vf commented:
Anything I can do to move this PR forward?

rpluem-vf commented:
Ready_for_review

samdoran reviewed (changes requested):
Sorry for not reviewing this sooner. I am not really happy with how this was solved in the `user` module and this seems to be further spreading that problematic solution. That being said, I can't think of a better way to solve it. 😞 

Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) and add integration tests and we can merge this. See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

samdoran commented:
Note to future @samdoran: I told you so. 😉

rpluem-vf committed:
Add changelog fragment for PR 59772

rpluem-vf commented:
I added a changelog fragment in d4176f279f2b25c326adea9c5701826275746e2e, but I have trouble to create a test. For a test I would need to create a group in a non local repository e.g. LDAP first and then try to create this group locally with the same gid. I am not sure how I should do this in an agnostic way such that the testing can be run at least on any Linux or even CentOS system.

rpluem-vf commented:
Ready_for_review

samdoran commented on changelogs/fragments/59772-fix_ansible_issue_58619.yaml:
```diff
@@ -0,0 +1,5 @@
+  - >
+     group - The group module was not correctly detecting whether a local
+     group is existing or not with local set to yes if the same group
+     exists in a non local group repository e.g. LDAP. Fix issue #58619.
```
```suggestion
     exists in a non local group repository e.g. LDAP. (https://github.com/ansible/ansible/issues/58619)
```

samdoran commented:
For the integration test, it's more about exercising the code path that handles the local file parsing. See the [user integration tests](https://github.com/ansible/ansible/blob/519672a6966aa5c187569d72d0bb5a3a510135eb/test/integration/targets/user/tasks/main.yml#L918-L918) for an example.

rpluem-vf committed:
Reference issue by URL and not by number

Co-Authored-By: Sam Doran <sdoran@redhat.com>

rpluem-vf committed:
Add test for PR 59772

rpluem-vf committed:
Do not check if the status has changed

If the test environment does not provide an additional repository for
groups on top of the local one then the second group creation task
will not change the state since the first task already, correctly,
created the group locally even with local: no.

rpluem-vf committed:
Only test on CentOS where 'lgroupadd' is installed
